lib_static/open_project/acts/registry_methods.rb

```diff
@@ -0,0 +1,58 @@
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2024 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module OpenProject
+  module Acts
+    module RegistryMethods
+      def instance(model_name)
+        class_name = model_name.singularize.camelize
+
+        class_name.constantize if class_names.include?(class_name)
+      end
+
+      def add(*models)
+        instance_methods_module = module_parent.const_get(:InstanceMethods)
+        acts_as_method_name = "acts_as_#{module_parent_name.demodulize.underscore}"
+
+        models.each do |model|
+          unless model.ancestors.include?(instance_methods_module)
+            raise ArgumentError.new("Model #{model} does not include #{acts_as_method_name}")
+          end
+
+          class_names << model.name
```

Comment:
Why did we change the behavior here to put the class names (as string) in the set and not the classes anymore? This requires us to use `constantize` in the `instance` method. When we still put the class names in the set, we can use the same `models.detect` method we used in the old implementation.

Reply:
I guess this is due to this: 

> Both registries were storing model classes, so in development new versions of those models were added when reloaded, but the registry still returned the first model class that was registered

But this could be fixed in the to_prepare call to simply clear out the registry.

Reply:
So you prefer a separate method to clear registry (or an argument of `add` method) over storing names of models? I'll have a look myself if it looks better

Reply:
Using Hash both resolves model reloading problem and doesn't need walking the set using `detect`, but also added explicit resetting of registries in `to_prepare`